src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -135,53 +169,125 @@ trait GraphDAO {
+          val workspaceVertex = getSinglePipelineResult(workspacePipeline(graph, workspaceName.namespace, workspaceName.name)).get
+          new GremlinPipeline(vertex).out(workflowEdge).linkIn(workflowEdge, workspaceVertex).iterate()
+        case Some(x: WorkflowFailure) => serializeDomainObjects(vertex, seq.asInstanceOf[Seq[WorkflowFailure]], workflowFailureEdge, (wf: WorkflowFailure) => (wf.entityType, wf.entityName), graph, workspaceName)
+      }
+      case (key, Some(mrConfig: MethodRepoConfiguration)) => serializeDomainObjects(vertex, Seq(mrConfig), methodRepoConfigEdge, thereShouldOnlyBeOne, graph, workspaceName)
+      case (key, Some(mrMethod: MethodRepoMethod)) => serializeDomainObjects(vertex, Seq(mrMethod), methodRepMethodEdge, thereShouldOnlyBeOne, graph, workspaceName)
+      case (key, Some(ws: WorkspaceName)) => // don't serialize workspace name objects
       case (key, Some(value)) => vertex.setProperty(key, value)
     })
 
-    vertex.setProperty("_clazz", obj.getClass.getSimpleName)
     vertex
   }
 
   /**
-   * Convert a graph vertex to an instance of T
-   * @param vertex
-   * @param otherProperties properties required to construct T but are not in the vertex. Keys must match the
-   *                        names of the fields on T to set
-   * @tparam T
-   * @return an instance of T populated from the vertex and otherProperties
+   * special function passed into idFxn of serializeDomainObjects when there should only be
+   * one object and thus any existing should be overwritten
    */
-  def fromVertex[T: TypeTag](vertex: Vertex, otherProperties: Map[String, Any]): T = {
-    fromPropertyMap(otherProperties ++ getVertexProperties(vertex))
+  private def thereShouldOnlyBeOne(x: Any) = 1
+
+  /**
+   * Serializes rawls domain objects to sub vertices. Will overwrite corresponding existing vertices, correspondence
+   * determined by idFxn.
+   *
+   * @param vertex the top level vertex
+   * @param objs the nested objects
+   * @param edgeLabel the label to give to the edges to the sub vertices
+   * @param idFxn function mapping an object of type T to an identifier, used to determine if vertices already in the
+   *              graph correspond to any of the nested objects
+   * @param graph used to create new vertices
+   * @param workspaceName used to construct domain objects that require a WorkspaceName
+   * @tparam T the type of the object being serializes
+   */
+  private def serializeDomainObjects[T: TypeTag: ClassTag](vertex: Vertex, objs: Seq[T], edgeLabel: String, idFxn: T => Any, graph: Graph, workspaceName: WorkspaceName): Seq[Vertex] = {
+    val existingObjVertexesById = vertex.getVertices(Direction.OUT, edgeLabel).map { objVertex =>
+      val obj = loadFromVertex[T](objVertex, Option(workspaceName))
+      idFxn(obj) -> objVertex
+    } toMap
+
+    (existingObjVertexesById -- objs.map(idFxn)).foreach(_._2.remove())
```

Comment:
comment would be good here - looks like to me removing the vertex in the map of existing vertexes for any vertex not in the new sub-graph